File: ruby/prime-factors/prime_factors.rb
class PrimeFactors
  def self.for(num)
    factors = []

    if num == 1
      return []
    end

    divisor = 2

    while num > 1
      if num%divisor == 0
        factors << divisor
        num /= divisor
        divisor = 2
      else
        divisor += 1
          if num%divisor == 0
            factors << divisor
            num /= divisor
          end
      end
    end

    factors
  end
end

File: javascript/prime-factors/prime-factors.js
var primeFactors = function(n){
  if(n === 1){
    return [];
  }
  
  var base = 2;
  var factors = [];

  while(n/base >= 1) {
    if(n%base === 0){
      n /= base;
      factors.push(base);
    } else {
      base++;
    }
  }

  return factors;
}

module.exports = { for: primeFactors };

